app/lib/ui/widgets/utility/pdf_screen.dart

```diff
@@ -71,33 +89,128 @@ class _PDFScreenState extends State<PDFScreen> {
         });
   }
 
-  // TODO: Give this function an object to build PDF from
-  Future<File> _getBuiltAndSavedPdf() async {
+  Future<File> _getBuiltAndSavedPdf(AnimalTransportRecord atr) async {
     final pdf = pw.Document();
+    pw.MemoryImage tranImage;
+    pw.MemoryImage shipperImage;
+    pw.MemoryImage receiverImage;
+    try {
+      final tranAckImage =
+          pw.MemoryImage(await networkImageToByte(atr.ackInfo.transporterAck));
+      setState(() {
+        tranImage = tranAckImage;
+      });
+    } catch (e) {
+      Text("Could not load transporter's  Acknowledgment Receipt");
+    }
+    try {
+      final shipperAckImage =
+          pw.MemoryImage(await networkImageToByte(atr.ackInfo.shipperAck));
+      setState(() {
+        shipperImage = shipperAckImage;
+      });
+    } catch (e) {
+      Text("Could not load transporter's  Acknowledgment Receipt");
```

Comment:
Having text here, and then not doing anything with it (meaning that the ack image could be null when given to `pw.Image` below) isn't right.

- [ ] Do `return Future.error("Could not load transporter's Acknowledgement Receipt")`, etc. instead